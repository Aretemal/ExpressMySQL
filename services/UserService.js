import Post from "../models/Post.js";
import User from "../models/User.js";

class UserService{
    async create(user){
        const createdPost = await User.create(user);
        return createdPost;
    }
    async getOne(id) {
        if (!id) {
            throw new Error('Id не указан')
        }
        const user = await User.findByPk(id);
        return user;
    }
    async getStatus(id) {
        if (!id) {
            throw new Error('Id не указан')
        }
        const user = await User.findByPk(id);
        if(!user){
            throw new Error('Пользователь не найден')
        }
        const status = user.status;
        return status;
    }
}
export default new UserService()
