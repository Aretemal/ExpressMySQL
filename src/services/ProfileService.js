import User from '../models/user.js';

class ProfileService {
  async getInfoAuthorizedUser(userId, next) {
    if (!userId) {
      next({ errorsArray: [{ msg: '441' }], title: 'Service Error' });
      return;
    }
    const user = await User.findByPk(userId);
    const {
      id, login, firstName, lastName, email, status, ava,
    } = user;
    return {
      id, login, firstName, lastName, email, status, ava,
    };
  }

  async updateStatus(status, userId, next) {
    if (!userId) {
      next({ errorsArray: [{ msg: '441' }], title: 'Service Error' });
      return;
    }
    const oldUser = await User.findByPk(userId);
    await oldUser.update({ status });
    const {
      id, login, firstName, lastName, email, ava,
    } = oldUser;
    return {
      id, login, firstName, lastName, email, status, ava,
    };
  }

  async getStatus(userId, next) {
    if (!userId) {
      next({ errorsArray: [{ msg: '440' }], title: 'Service Error' });
      return;
    }
    const user = await User.findByPk(userId);
    if (!user) {
      next({ errorsArray: [{ msg: '440' }], title: 'Service Error' });
      return;
    }
    const {
      id, login, firstName, lastName, email, ava, status,
    } = user;
    return {
      id, login, firstName, lastName, email, ava, status,
    };
  }
}
export default new ProfileService();
