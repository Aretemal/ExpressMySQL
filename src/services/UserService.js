import User from '../models/user.js';
import { ResponseObjectJSON } from '../utils/creatorJSON.js';

class UserService {
  async getOne(id) {
    if (!id) {
      throw new Error('Id не указан');
    }
    const user = await User.findByPk(id);
    return ResponseObjectJSON.render({
      user,
      resultCode: 0,
    });
  }

  async getStatus(id) {
    if (!id) {
      throw new Error('Id не указан');
    }
    const user = await User.findByPk(id);
    if (!user) {
      throw new Error('Пользователь не найден');
    }
    const { status } = user;
    return ResponseObjectJSON.render({
      status,
      resultCode: 0,
    });
  }
}
export default new UserService();
