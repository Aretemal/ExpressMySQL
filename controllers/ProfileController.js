import ProfileService from '../services/ProfileService.js'
class ProfileController{
    async getInfoAuthorizedUser(req, res){
        try {
            const user = await ProfileService.getInfoAuthorizedUser(req.user.id);
            return res.json(user);
        } catch (e){
            res.status(500).json(e)
        }
    }
    async updateStatus(req, res){
        try {
            const user = await ProfileService.updateStatus(req.body.status, req.user.id);
            return res.json({resultCode: 0});
        } catch (e){
            res.status(500).json(e)
        }
    }
}
export default new ProfileController()
