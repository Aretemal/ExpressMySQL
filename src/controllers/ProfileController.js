import ProfileService from '../services/ProfileService.js';
import fullUrlCreator from '../utils/fullUrlCreator.js';
import ProfileSerializers from '../serializers/ProfileSerializers.js';

class ProfileController {
  async getInfoAuthorizedUser(req, res, next) {
    const data = await ProfileService.getInfoAuthorizedUser(req.user.id);
    const serializer = new ProfileSerializers({
      attributes: data, id: req.user.id, type: 'User info', link: fullUrlCreator(req),
    });
    req.serializer = serializer;
    next();
  }

  async updateStatus(req, res, next) {
    const data = await ProfileService.updateStatus(req.body.status, req.user.id);
    const serializer = new ProfileSerializers({
      attributes: data, id: req.user.id, type: 'New status', link: fullUrlCreator(req),
    });
    req.serializer = serializer;
    next();
  }

  async getStatus(req, res, next) {
    const status = await ProfileService.getStatus(req.params.id);
    const serializer = new ProfileSerializers({
      attributes: status, id: req.user.id, type: 'Status', link: fullUrlCreator(req),
    });
    req.serializer = serializer;
    next();
  }
}
export default new ProfileController();
