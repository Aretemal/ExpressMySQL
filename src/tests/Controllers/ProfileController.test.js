/* eslint-disable no-undef */
import { jest } from '@jest/globals';
import ProfileController from '../../controllers/ProfileController';
import ProfileService from '../../services/ProfileService';
import UserService from '../../services/UserService.js';

jest.fn('ProfileService');
const req = {
  params: {
    id: 1,
  },
  user: {
    id: 1,
  },
  body: {
    status: 'Status',
  },
};
const res = {
  dataJS: null,
  json: (data) => {
    res.dataJS = data;
  },
};

describe('Profile Controller :', () => {
  describe('getInfoAuthorizedUser :', () => {
    test('should return info authorized user', async () => {
      function next() {}
      ProfileService.getInfoAuthorizedUser = (id) => ({
        userId: id, login: 'Artem', firstName: 'Artem', lastName: 'Novik',
      });

      await ProfileController.getInfoAuthorizedUser(req, res, next);

      expect(req.serializer).toMatchSnapshot();
    });
  });

  describe('updateStatus :', () => {
    test('should return user', async () => {
      function next() {}
      ProfileService.updateStatus = (id) => ({
        userId: id, login: 'Artem', firstName: 'Artem', lastName: 'Novik',
      });

      await ProfileController.updateStatus(req, res, next);

      expect(req.serializer).toMatchSnapshot();
    });
  });

  describe('getStatus :', () => {
    test('should return user', async () => {
      function next() {}
      UserService.getStatus = (id) => ({
        userId: id, login: 'Artem', firstName: 'Artem', lastName: 'Novik',
      });

      await ProfileController.getStatus(req, res, next);

      expect(req.serializer).toMatchSnapshot();
    });
  });
});
