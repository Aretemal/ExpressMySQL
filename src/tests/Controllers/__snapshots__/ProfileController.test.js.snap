// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile Controller : getInfoAuthorizedUser : should return info authorized user 1`] = `
UserSerializer {
  "resource": {
    "firstName": "Artem",
    "lastName": "Novik",
    "login": "Artem",
    "userId": 1,
  },
}
`;

exports[`Profile Controller : getStatus : should return user 1`] = `
UserSerializer {
  "resource": {
    "ava": null,
    "email": "user1",
    "firstName": "user1",
    "id": 1,
    "lastName": "user1",
    "login": "user1",
    "status": "User",
  },
}
`;

exports[`Profile Controller : updateStatus : should return user 1`] = `
UserSerializer {
  "resource": {
    "firstName": "Artem",
    "lastName": "Novik",
    "login": "Artem",
    "userId": "Status",
  },
}
`;
